domino/domino.py

```diff
@@ -22,26 +20,35 @@
 
 
 class Domino:
-    def __init__(self, project, api_key=None, host=None, domino_token_file=None):
+    def __init__(self, project, api_key=None, host=None, domino_token_file=None, auth_token=None):
+
         self._configure_logging()
-        host = clean_host_url(get_host_or_throw_exception(host))
-        domino_token_file = get_path_to_domino_token_file(domino_token_file)
-        api_key = get_api_key(api_key)
 
-        # Initialise request manager
-        self.request_manager = self._initialise_request_manager(api_key, domino_token_file)
+        _host = host or os.getenv(DOMINO_HOST_KEY_NAME)
+        assert _host, ("Host must be supplied as a parameter or through the "
+                       f"{DOMINO_HOST_KEY_NAME} environment variable.")
+        host = clean_host_url(_host)
+
+        try:
+            owner_username, project_name = project.split("/")
+            self._routes = _Routes(host, owner_username, project_name)
+        except ValueError as e:
+            self._logger.error(f"Project {project} must be given in the form username/projectname")
+            raise
+
+        domino_token_file = domino_token_file or os.getenv(DOMINO_TOKEN_FILE_KEY_NAME)
+        api_key = api_key or os.getenv(DOMINO_USER_API_KEY_KEY_NAME)
 
-        owner_username, project_name = project.split("/")
-        self._routes = _Routes(host, owner_username, project_name)
+        self.request_manager = self._authenticate(api_key, auth_token, domino_token_file)
```

Comment:
Related to the comment re: `_authenticate()` removal below.
```suggestion
        self.request_manager = _HttpRequestManager(get_auth_by_type(api_key=api_key, auth_token=auth_token, domino_token_file=domino_token_file))
```

Reply:
Done.